src/create.c

```diff
@@ -74,24 +78,45 @@ int main(int argc, char **argv) {
     singularity_message(INFO, "Creating %ldMiB image\n", size);
     singularity_image_create(&image, size);
 
+#ifdef SUID_CREATE
     singularity_message(INFO, "Binding image to loop\n");
     singularity_image_bind(&image);
+#endif
 
     mkfs_cmd[0] = strdup("/sbin/mkfs.ext3");
     mkfs_cmd[1] = strdup("-q");
+
+#ifdef SUID_CREATE
     mkfs_cmd[2] = strdup(image.loopdev);
     mkfs_cmd[3] = NULL;
+#else
+    mkfs_cmd[2] = strdup("-E");
+    // the offset in the file for the singularity header
+    mkfs_cmd[3] = strjoin("offset=", int2str(strlength(LAUNCH_STRING, 1024)));
+    mkfs_cmd[4] = strdup(image.path);
+    // pass the correct size of the file in KiB
+    mkfs_cmd[5] = int2str((size*1024*1024-strlength(LAUNCH_STRING, 1024))/1024);
```

Comment:
If you don't pass it, you get (even with `-q`):
```
Warning: offset specified without an explicit file system size.
Creating a file system with 131072 blocks but this might
not be what you want.
```
So I think it's safer to add it.